### ISSUE TITLE
Build failure with SDL 1.x

### ISSUE THREAD

**Issue by alexey-lysiuk** [gerstrong/Commander-Genius#224]
The following compilation errors prevent building without `USE_SDL2`:
```
lib/GsKit/base/video/COpenGL.cpp:222:27: error: use of undeclared identifier 'oglfilter'
        createTexture(m_texture, oglfilter, m_GamePOTScaleDim.w, m_GamePOTScaleDim.h);
                                 ^
lib/GsKit/base/video/COpenGL.cpp:226:27: error: use of undeclared identifier 'oglfilter'
          createTexture(m_texFX, oglfilter, m_GamePOTScaleDim.w, m_GamePOTScaleDim.h, true);
                                 ^
src/engine/core/VGamepads/vgamepadsimple.cpp:101:9: error: use of undeclared identifier 'mDPadTexture'
        mDPadTexture.setAlpha(uint8_t(255.0f*mTranslucency));
        ^
src/engine/core/VGamepads/vgamepadsimple.cpp:102:22: error: no member named 'addTextureRefToRender' in 'CVideoDriver'
        gVideoDriver.addTextureRefToRender(mDPadTexture, dpadRect);
        ~~~~~~~~~~~~ ^
src/engine/core/VGamepads/vgamepadsimple.cpp:102:44: error: use of undeclared identifier 'mDPadTexture'
        gVideoDriver.addTextureRefToRender(mDPadTexture, dpadRect);
                                           ^
src/engine/core/VGamepads/vgamepadsimple.cpp:114:9: error: use of undeclared identifier 'mConfirmButtonTexture'
        mConfirmButtonTexture.setAlpha(uint8_t(255.0f*mTranslucency));
        ^
src/engine/core/VGamepads/vgamepadsimple.cpp:115:22: error: no member named 'addTextureRefToRender' in 'CVideoDriver'
        gVideoDriver.addTextureRefToRender(mConfirmButtonTexture, confirmRect);
        ~~~~~~~~~~~~ ^
src/engine/core/VGamepads/vgamepadsimple.cpp:115:44: error: use of undeclared identifier 'mConfirmButtonTexture'
        gVideoDriver.addTextureRefToRender(mConfirmButtonTexture, confirmRect);
                                           ^
src/engine/core/VGamepads/vgamepadsimple.cpp:127:9: error: use of undeclared identifier 'mStartButtonTexture'
        mStartButtonTexture.setAlpha(uint8_t(255.0f*mTranslucency));
        ^
src/engine/core/VGamepads/vgamepadsimple.cpp:128:22: error: no member named 'addTextureRefToRender' in 'CVideoDriver'
        gVideoDriver.addTextureRefToRender(mStartButtonTexture, confirmRect);
        ~~~~~~~~~~~~ ^
src/engine/core/VGamepads/vgamepadsimple.cpp:128:44: error: use of undeclared identifier 'mStartButtonTexture'
        gVideoDriver.addTextureRefToRender(mStartButtonTexture, confirmRect);
                                           ^
src/engine/core/VGamepads/vgamepadsimple.cpp:140:9: error: use of undeclared identifier 'mShootButtonTexture'
        mShootButtonTexture.setAlpha(uint8_t(255.0f*mTranslucency));
        ^
src/engine/core/VGamepads/vgamepadsimple.cpp:141:22: error: no member named 'addTextureRefToRender' in 'CVideoDriver'
        gVideoDriver.addTextureRefToRender(mShootButtonTexture, shootButtonRect);
        ~~~~~~~~~~~~ ^
src/engine/core/VGamepads/vgamepadsimple.cpp:141:44: error: use of undeclared identifier 'mShootButtonTexture'
        gVideoDriver.addTextureRefToRender(mShootButtonTexture, shootButtonRect);
                                           ^
src/engine/core/VGamepads/vgamepadsimple.cpp:144:9: error: use of undeclared identifier 'mJumpButtonTexture'
        mJumpButtonTexture.setAlpha(uint8_t(255.0f*mTranslucency));
        ^
src/engine/core/VGamepads/vgamepadsimple.cpp:145:22: error: no member named 'addTextureRefToRender' in 'CVideoDriver'
        gVideoDriver.addTextureRefToRender(mJumpButtonTexture, jumpButtonRect);
        ~~~~~~~~~~~~ ^
src/engine/core/VGamepads/vgamepadsimple.cpp:145:44: error: use of undeclared identifier 'mJumpButtonTexture'
        gVideoDriver.addTextureRefToRender(mJumpButtonTexture, jumpButtonRect);
                                           ^
src/engine/core/VGamepads/vgamepadsimple.cpp:148:9: error: use of undeclared identifier 'mPogoButtonTexture'
        mPogoButtonTexture.setAlpha(uint8_t(255.0f*mTranslucency));
        ^
src/engine/core/VGamepads/vgamepadsimple.cpp:149:22: error: no member named 'addTextureRefToRender' in 'CVideoDriver'
        gVideoDriver.addTextureRefToRender(mPogoButtonTexture, pogoButtonRect);
        ~~~~~~~~~~~~ ^
src/engine/core/VGamepads/vgamepadsimple.cpp:149:44: error: use of undeclared identifier 'mPogoButtonTexture'
        gVideoDriver.addTextureRefToRender(mPogoButtonTexture, pogoButtonRect);
                                           ^
src/engine/core/VGamepads/vgamepadsimple.cpp:152:9: error: use of undeclared identifier 'mStatusButtonTexture'
        mStatusButtonTexture.setAlpha(uint8_t(255.0f*mTranslucency));
        ^
```
Errors in `COpenGL.cpp` are trivial to fix, not sure about those in `VirtualKeenControl` class.

Is SDL 1.x still supported?

**Comment by alexey-lysiuk** [comment]
Also, the related code in `COpenGL.cpp` looks suspicious:
```cpp
    GLint oglfilter = GL_LINEAR;
    if(m_VidConfig.mRenderScQuality == "nearest")
        const GLint oglfilter = GL_NEAREST;
```
IMHO `const GLint` in the last line was added by mistake.

**Comment by gerstrong** [comment]
Yes that must be incorrect

Gesendet von meinem ASUS MeMO Pad

**Comment by gerstrong** [comment]
Sdl 1.2 is still supported, yes and it will continue to be. Vgamepad class should be disabled in most parts. It is meant to be used with sdl 2.0 only. For what i see i need to fix some stuff in the code...

Gesendet von meinem ASUS MeMO Pad

**Comment by gerstrong** [comment]
Fixed since 32e62216dd8ab45c08e9de1e0d395a00d5d4979e
